dspace-api/src/test/java/org/dspace/statistics/export/processor/BitstreamEventProcessorTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.statistics.export.processor;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.UnsupportedEncodingException;
+import java.util.UUID;
+
+import org.dspace.AbstractDSpaceTest;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Item;
+import org.dspace.services.ConfigurationService;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+/**
+ * Test class for the BitstreamEventProcessor
+ */
+public class BitstreamEventProcessorTest extends AbstractDSpaceTest {
+
+    @Mock
+    private Item item = mock(Item.class);
+
+    @Mock
+    private Bitstream bitstream = mock(Bitstream.class);
+
+    @Mock
+    private ConfigurationService configurationService = mock(ConfigurationService.class);
+
+    @InjectMocks
+    BitstreamEventProcessor bitstreamEventProcessor = mock(BitstreamEventProcessor.class, CALLS_REAL_METHODS);
+
+    @Test
+    /**
+     * Test the method that adds data based on the object types
+     */
+    public void testAddObectSpecificData() throws UnsupportedEncodingException {
+        bitstreamEventProcessor.configurationService = configurationService;
+        when(configurationService.getProperty(any(String.class))).thenReturn("demo.dspace.org");
```

Comment:
Just a note: You don't need to mock a config setting. You can just set it using configurationService.setProperty().  Configs reload after every test.

Reply:
Removed the mocking of these properties, and set them in the class instead